code/game/machinery/computer/law.dm

```diff
@@ -36,9 +46,45 @@
 		if(!atoms_share_level(T, src))
 			to_chat(user, "<span class='danger'>Unable to establish a connection</span>: You're too far away from the target silicon!")
 			return
-		var/obj/item/aiModule/M = O
-		M.install(src)
-		return
+		if (!emagged) //non-emag law change
+			var/obj/item/aiModule/M = O
+			M.install(src)
+			return
+		else
+			if(world.time < cooldown)
+				to_chat(user, "The program seems to have frozen. It will need some time to process")
+				return
+			do_sparks(5, 1, src)
+			var/foundlaws = 0
+			var/checked = FALSE
+			for(var/datum/ai_law/law in current.laws.all_laws())
+				if (law in current.laws.inherent_laws)
+					foundlaws = foundlaws + 1
+				if(current.laws.ion_laws.len == 0 && checked == FALSE)
+					checked = TRUE
+					if (rand(1,10) >= 9)  // 20% chance to generate an ion law if none exists
+						current.add_ion_law(generate_ion_law())
+						cooldown = world.time + 600
+						return
+				else if(law in current.laws.ion_laws && checked == FALSE) //10% chance to overwrite a current ion
+					checked = TRUE
+					if (rand(1,10) >= 10)
+						current.clear_ion_laws()
+						current.add_ion_law()
+						cooldown = world.time + 600
+						return
+			var/lawposition = rand(1,foundlaws)
+			if (foundlaws != 0) //as long as it finds a law to change
+				current.laws.inherent_laws[lawposition].law = new/datum/ai_law/inherent(generate_ion_law()).law
+				log_and_message_admins("has given [src] uploaded the emag'd inherent law: [current.laws.inherent_laws[lawposition].law]")
+				current.show_laws()
+				current.throw_alert("newlaw", /obj/screen/alert/newlaw)
+				cooldown = world.time + 600
+				var/i
+				for (i=1,i<=current.connected_robots.len,i++) // push alert to the AI's borgs
```

Comment:
```suggestion
				for(var/i in 1 to length(current.connected_robots)) // push alert to the AI's borgs
```
We prefer not to use this style of for. I think it could work like this (76% sure)